deploy/panos.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env sh
+
+# Script to deploy certificates to Palo Alto Networks PANOS via API
+# Note PANOS API KEY and IP address needs to be set prior to running.
+# The following variables exported from environment will be used.
+# If not set then values previously saved in domain.conf file are used.
+#
+# Firewall admin with superuser and IP address is required.
+#
+# export PANOS_USER=""  # required
+# export PANOS_PASS=""  # required
+# export PANOS_HOST=""  # required
+
+# This function is to parse the XML
+parse_response() {
+  status=$(echo "$1" | sed 's/^.*"\([a-z]*\)".*/\1/g')
+  message=$(echo "$1" | sed 's/^.*<result>\(.*\)<\/result.*/\1/g')
+  return 0
+}
+
+# This function is to deploy the cert to the Firewall
+deploy_cert() {
+  #Checking for IP and Key
+  if [ -z "$_panos_host" ] && [ -z "$_panos_key" ]; then
+    _err "No IP or Key found. Can not deploy certs."
+    return 1
+  else # IP and key found lets import cert, key and commit.
+    _debug "Deploying Cert"
+    cert_response=$(curl -k --form file=@"$_cfullchain" "https://$_panos_host/api/?type=import&category=certificate&certificate-name=$_cdomain&format=pem&key=$_panos_key")
```

Comment:
don't use `curl`,  please use `_get()` or `_post()` function instead.
we need to support wget.

Reply:
@Neilpang How do I make a form-data file post using your  _post() function?

Reply:
it's submitting a form to upload cert and key.  we can use `_post()` to simulate this action.
It's just about the content-type and the postbody.

content-type is just a header, which is easy to set.
postbody is urlencoded file,  which is  percentage sign`%` splited hex encoding bytes.

We have a `_url_encode()` function to do so.

Reply:
Thanks Neil refactored the code to use _post